apps/coordinator/src/selectors/wallet.ts

```diff
@@ -0,0 +1,480 @@
+ * where the state of that slice is not known
+ */
+export const getUnknownAddresses = createSelector(
+  [getWalletSlices, getUnknownAddressSlices],
+  (slices: Slice[]): string[] =>
+    slices.map((slice: Slice) => slice.multisig.address),
+);
+
+/**
+ * @description Returns a selector of all slices from the _deposit_ braid
+ * where the address hasn't been used yet.
+ */
+export const getDepositableSlices = createSelector(
+  getDepositSlices,
+  (slices: Slice[]): Slice[] =>
+    slices.filter(
+      (slice: Slice) => slice.balanceSats.isEqualTo(0) && !slice.addressUsed,
+    ),
+);
+
+/**
+ * @description Returns a selector of the text needed to construct a wallet
+ * details file.
+ */
+export const getWalletDetailsText = createSelector(
+  [
+    getWalletName,
+    getWalletUuid,
+    getAddressType,
+    getNetwork,
+    getClientDetails,
+    getRequiredSigners,
+    getTotalSigners,
+    getExtendedPublicKeysBIP32Paths,
+    getStartingAddressIndex,
+    getWalletLedgerPolicyHmacs,
+  ],
+  (
+    walletName: string,
+    walletUuid: string,
+    addressType: string,
+    network: string,
+    clientDetails: string,
+    requiredSigners: number,
+    totalSigners: number,
+    extendedPublicKeys: string,
+    startingAddressIndex: number,
+    ledgerPolicyHmacs: LedgerPolicyHmac[] = [],
+  ): string => {
+    return `{
+  "name": "${walletName}",
+  "uuid": "${walletUuid}",
+  "addressType": "${addressType}",
+  "network": "${network}",
+  "client":  ${clientDetails},
+  "quorum": {
+    "requiredSigners": ${requiredSigners},
+    "totalSigners": ${totalSigners}
+  },
+  "extendedPublicKeys": [
+    ${extendedPublicKeys}
+  ],
+  "startingAddressIndex": ${startingAddressIndex},
+  "ledgerPolicyHmacs": [${ledgerPolicyHmacs.map((hmac: LedgerPolicyHmac) => JSON.stringify(hmac)).join(", ")}]
+}`;
+  },
+);
+
+export const getWalletConfig = createSelector(
+  [getWalletDetailsText],
+  (walletDetailsText: string) => JSON.parse(walletDetailsText),
+);
+
+interface HmacWithName {
+  policyHmac: string;
+  name: string;
+}
+
+export const getHmacsWithName = createSelector(
+  getExtendedPublicKeyImporters,
+  getWalletLedgerPolicyHmacs,
+  (
+    extendedPublicKeys: Record<string, ExtendedPublicKeyImporter>,
+    policyHmacs: LedgerPolicyHmac[],
+  ): HmacWithName[] => {
+    return Object.values(extendedPublicKeys)
+      .map((importer: ExtendedPublicKeyImporter) => {
+        const policyHmac = policyHmacs.find(
+          (hmac: LedgerPolicyHmac) => hmac.xfp === importer.rootXfp,
+        )?.policyHmac;
+        return { policyHmac, name: importer.name };
+      })
+      .filter(
+        (registration: {
+          policyHmac?: string;
+          name: string;
+        }): registration is HmacWithName => Boolean(registration.policyHmac),
+      );
+  },
+);
+
+/**
+ * @description Returns an array of all wallet addresses from both deposit and change nodes
+ */
+export const getWalletAddresses = createSelector(
```

Comment:
everything below this is new to the file. This turns some of the utility code from the old hooks file into selectors to get memoization out of the box without useMemo or useCallback